### PR TITLE
Fix python3 syntax in rocky

### PR DESCRIPTION
**Pull request by grandchild**
Two little syntax issues with the current rocky code.

There are 60+ unit tests failing for me as well...

### CODE REVIEW

**Comment by grandchild** [comment]
I found these when running
```
tox -e py36
```

The syntax errors are a bit more easy to read with
```
tox -e py36 |& sed s/\\\\n/\\n/g
```

**Review by mariusleu — APPROVED** [review]
Thanks for this.
We're not targeting python3 now (the next upgrade will) but it's good to get our patches ready for it.

File: nova/virt/vmwareapi/special_spawning.py [hunk 1]
```diff
@@ -257,7 +257,7 @@ def free_host(self, context):
                             for h, vms in vms_per_host.items()}
 
             # take the one with least memory used
-            host, _ = sorted(mem_per_host.items(), key=lambda (x, y): y)[0]
+            host, _ = sorted(mem_per_host.items(), key=lambda x, y: y)[0]
```
Comment (joker-official) [inline nova/virt/vmwareapi/special_spawning.py:260]: This breaks on python2:
```
$ python3
Python 3.5.2 (default, Oct  7 2020, 17:19:02) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> mem_per_host = {}
>>> ost, _ = sorted(mem_per_host.items(), key=lambda (x, y): y)[0]
  File "<stdin>", line 1
    ost, _ = sorted(mem_per_host.items(), key=lambda (x, y): y)[0]
                                                     ^
SyntaxError: invalid syntax
>>> 
```
```
$ python2
Python 2.7.12 (default, Oct  5 2020, 13:56:01) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> mem_per_host = {}
>>> ost, _ = sorted(mem_per_host.items(), key=lambda (x, y): y)[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: list index out of range
>>> 
```
```
$ python3
Python 3.5.2 (default, Oct  7 2020, 17:19:02) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> mem_per_host = {}
>>> host, _ = sorted(mem_per_host.items(), key=lambda x, y: y)[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: list index out of range
>>> 
```
```
$ python2
Python 2.7.12 (default, Oct  5 2020, 13:56:01) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> mem_per_host = {}
>>> host, _ = sorted(mem_per_host.items(), key=lambda x, y: y)[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: list index out of range
>>> mem_per_host = {1: 2, 3: 4}
>>> host, _ = sorted(mem_per_host.items(), key=lambda x, y: y)[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: <lambda>() takes exactly 2 arguments (1 given)

**Review by joker-official — CHANGES_REQUESTED** [review]

**Comment by joker-official** [comment]
We need to fix the pep8 check ... :/

File: nova/virt/vmwareapi/special_spawning.py (hunk 1 above)
Comment (grandchild) [reply nova/virt/vmwareapi/special_spawning.py:260]: ah it's not just two args, it's tuple unpacking. yeah, that's gone as well in py3... hm.

Comment (joker-official) [reply nova/virt/vmwareapi/special_spawning.py:260]: we could use `attrgetter(1)` I guess

Comment (joker-official) [reply nova/virt/vmwareapi/special_spawning.py:260]: I mean `itemgetter(1)`:
```
>>> from operator import itemgetter
>>> host, _ = sorted(mem_per_host.items(), key=itemgetter(1))
```

**Comment by grandchild** [comment]
> We're not targeting python3 now

Ah, I had that wrong!

File: nova/virt/vmwareapi/special_spawning.py [hunk 2]
```diff
@@ -257,7 +257,7 @@ def free_host(self, context):
                             for h, vms in vms_per_host.items()}
 
             # take the one with least memory used
-            host, _ = sorted(mem_per_host.items(), key=lambda (x, y): y)[0]
+            host, _ = sorted(mem_per_host.items(), key=lambda h: h[1])[0]
```
Comment (joker-official) [inline nova/virt/vmwareapi/special_spawning.py:260]: works, too, but getting rid of `lambda` is nicer imho

**Comment by joker-official** [comment]
Could you rebase please, so we can see it fixes the pep8 check, too?

**Comment by grandchild** [comment]
Whoops, missed the comment two weeks ago. Rebased now.

**Comment by grandchild** [comment]
From https://ci.eu-de-2.cloud.sap/teams/core/pipelines/pull-requests/jobs/nova-tox/builds/135:
```
pep8 run-test: commands[3] | bash tools/check-cherry-picks.sh
Stable branch requires either cherry-pick -x headers or [stable-only] tag!
ERROR: InvocationError for command /bin/bash tools/check-cherry-picks.sh (exited with code 1)
```

**Pull request merged**